search/search_quality/assessment_tool/main_view.cpp

```diff
@@ -58,38 +60,63 @@ void MainView::SetSamples(ContextList::SamplesSlice const & samples)
   m_sampleView->Clear();
 }
 
-void MainView::ShowSample(size_t index, search::Sample const & sample, bool hasEdits)
+void MainView::ShowSample(size_t sampleIndex, search::Sample const & sample, bool hasEdits)
 {
   MoveViewportToRect(sample.m_viewport);
 
   m_sampleView->SetContents(sample);
   m_sampleView->show();
 
-  OnSampleChanged(index, Edits::Update::AllRelevancesUpdate(), hasEdits);
+  OnResultChanged(sampleIndex, ResultType::Found, Edits::Update::AllRelevancesUpdate());
+  OnResultChanged(sampleIndex, ResultType::NonFound, Edits::Update::AllRelevancesUpdate());
+  OnSampleChanged(sampleIndex, hasEdits);
+}
+
+void MainView::ShowFoundResults(search::Results::Iter begin, search::Results::Iter end)
+{
+  m_sampleView->ShowFoundResults(begin, end);
 }
 
-void MainView::ShowResults(search::Results::Iter begin, search::Results::Iter end)
+void MainView::ShowNonFoundResults(std::vector<search::Sample::Result> const & results)
 {
-  m_sampleView->ShowResults(begin, end);
+  m_sampleView->ShowNonFoundResults(results);
 }
 
 void MainView::MoveViewportToResult(search::Result const & result)
 {
   m_framework.ShowSearchResult(result, false /* animation */);
 }
 
+void MainView::MoveViewportToResult(search::Sample::Result const & result) {
+  int constexpr kViewportAroundResultSizeM = 100;
```

Comment:
Nothing wrong. Just MercatorBounds::RectByCenterXYAndSizeInMeters() accepts double. It's up to you.